v2/http-middleware/security.go

```diff
@@ -240,6 +249,27 @@ func AuthorizeMiddleware(authorizer Authorizer) mux.MiddlewareFunc {
 	}
 }
 
+type ResourceFuncError struct {
```

Comment:
This is more or less HTTPError. Sure, it is only used for resourceFunc error for now, but it could just as much be used for all http errors :) Suggestion is renaming it.